BepInEx.Core/Bootstrap/TypeLoader.cs

```diff
@@ -10,6 +10,47 @@
 
 namespace BepInEx.Bootstrap;
 
+internal class CachedPluginLoadContext : IPluginLoadContext, IDisposable
+{
+    public IPluginLoadContext PluginLoadContext { get; }
+    private byte[] assemblyData;
+    private byte[] assemblySymbolsData;
+    private readonly Dictionary<string, byte[]> files = new();
+
+    public CachedPluginLoadContext(IPluginLoadContext pluginLoadContext)
+    {
+        PluginLoadContext = pluginLoadContext;
+    }
+
+    public string AssemblyIdentifier => PluginLoadContext.AssemblyIdentifier;
+    public string AssemblyHash => PluginLoadContext.AssemblyHash;
+    public byte[] GetAssemblyData()
+    {
+        return assemblyData ??= PluginLoadContext.GetAssemblyData();
+    }
+
+    public byte[] GetAssemblySymbolsData()
+    {
+        return assemblySymbolsData ??= PluginLoadContext.GetAssemblySymbolsData();
+    }
+
+    public byte[] GetFile(string relativePath)
+    {
+        if (!files.ContainsKey(relativePath))
+        {
+            files[relativePath] = PluginLoadContext.GetFile(relativePath);
+        }
+        return files[relativePath];
+    }
```

Comment:
This might turn into a memory hog if plugins try to open large asset files this way. Also if the paths are case-insensitive this could hold multiple copies of the same file. Maybe it should only cache files below some size?

Reply:
I was heavily debating on having files caching in the first place, but I hadn't considered these cases so I removed it as plugins could decide themselves to do it if they really want it.